sympy/polys/matrices/domainmatrix.py

```diff
@@ -21,8 +21,72 @@
 
 
 class DomainMatrix:
+    r"""
+    Associate Matrix with :py:class:`~.Domain`
+
+    Explanation
+    ===========
+
+    DomainMatrix uses "domain elements" for its internal representation
+    which makes it more faster for many common operations
+    than current sympy Matrix class, but this advantage makes it not
+    entirely compatible with Matrix.
+    DomainMatrix could be found analogous to numpy arrays with "dtype".
+    In the DomainMatrix, each matrix has a domain such as ZZ(Integers)
+    or QQ<sqrt(2)>[Algebraic Number Field].
+
+
+    Examples
+    ========
+
+    Creating a DomainMatrix from the existing Matrix class:
+
+    >>> from sympy import Matrix
+    >>> from sympy.polys.matrices import DomainMatrix
+    >>> Matrix1 = Matrix([
+    ...    [1, 2],
+    ...    [3, 4]])
+    >>> A = DomainMatrix.from_Matrix(Matrix1)
+    >>> A
+    DomainMatrix([[1, 2], [3, 4]], (2, 2), ZZ)
+
+    Driectly forming a DomainMatrix:
+
+    >>> from sympy import ZZ
+    >>> from sympy.polys.matrices import DomainMatrix
+    >>> A = DomainMatrix([
+    ...    [ZZ(1), ZZ(2)],
+    ...    [ZZ(3), ZZ(4)]], (2, 2), ZZ)
+    >>> A
+    DomainMatrix([[1, 2], [3, 4]], (2, 2), ZZ)
+
+    See Also
+    ========
+
+    :py:class:`~.ddm`
+    :py:class:`~.Domain`
+    :py:class:`~.Poly`
```

Comment:
I'm not sure this is the right syntax for the see also section. If these were added to the rst docs then I think this might fail.

Reply:
You were right. This is failing when added to `rst` file. I'll fix it